### ISSUE TITLE
Request UI updates from debug adapter implementation

### ISSUE THREAD

**Issue by theIDinside** [microsoft/vscode#141311]
I want to be able to request VSCode to perform all the various requests. For instance, in the debug protocol, the UI only ever updates after it's chain of `...Request` calls. However, due to the inherent async nature of debugging, things might change behind the scenes, which is entirely impossible, the way VSCode works now, to update in the UI.

One very concrete example is this:
A user has a set of breakpoints in the breakpoints list, some of which might be "unverified" (for instance, when debugging a C++ program, the library might not have been loading, and thus, the breakpoint is "pending"). However, after the breakpoint is hit, as a debug adapter implementer, I want to be able to tell VSCode that "hey, this breakpoint is actually verified now" but there is *no* way for me to do so.

One way, to provide features like this would be:
A: (preferable) be able to trigger a request chain, or a specific request. The best thing would be able to ask VSCode (from the debug adapter's perspective) to request *one* specific request, and not be drowned in unnecessary overhead by triggering an entire request chain again.

B: give direct control to the Debug UI to the debug adapter, although I think this would probably require a lot more work. 

So, in summary:
A debug adapter implementer should be able to ask VSCode to fire a request (which in the end, the debug adapter will handle, since it is the one implementing it).

**Comment by theIDinside** [comment]
I did for instance, in the case of the mid-flight verification of breakpoints, `vscode.debug.removeBreakpoints(...)` on the unverified and then `vscode.debug.addBreakpoints(...)` - but the problem with this, is that it triggers the setBreakpointsRequest i think, so maybe the A in my A / B list is not a desirable solution after all, since we won't know when or who sent the request, and if the target (inferior / debugee / tracee) is running, can throw errors.

Therefore B; giving direct control of the debug UI:t possibly is the only way forward.

**Comment by weinand** [comment]
@theIDinside you said:
> A user has a set of breakpoints in the breakpoints list, some of which might be "unverified" (for instance, when debugging a C++ program, the library might not have been loading, and thus, the breakpoint is "pending"). However, after the breakpoint is hit, as a debug adapter implementer, I want to be able to tell VSCode that "hey, this breakpoint is actually verified now" but there is no way for me to do so.

This can be done by sending a DAP `Breakpoint` (see https://microsoft.github.io/debug-adapter-protocol/specification#Events_Breakpoint). 

Mock Debug has an example for this: https://github.com/microsoft/vscode-mock-debug/blob/7147a68373309fc9789e99a629a13d1a27222d30/src/mockRuntime.ts#L654-L660
